mmv1/third_party/terraform/utils/config.go.erb

```diff
@@ -194,6 +194,41 @@ func (c *Config) LoadAndValidate(ctx context.Context) error {
 	c.clientEventarcDCL = eventarcDcl.NewClient(c.dclConfig)
 <% end -%>
 
+	// Print Identities executing terraform API Calls.
+
+	if c.ImpersonateServiceAccount != "" {
+		// Drop Impersonated ClientOption from OAuth2 TokenSource to infer original identity
+
+		tokenSource, err = c.getTokenSource(c.Scopes, true)
+		if err != nil {
+			return err
+		}
+		client = oauth2.NewClient(cleanCtx, tokenSource)
+
+		email, err := GetCurrentUserEmail(c, c.userAgent)
+		if err != nil {
+			return err
```

Comment:
Since we don't know whether the provided scopes will include the email scope, we should log a message on an error instead of returning.